ios/RNKiwiMobile/RNKiwiViewController.m

```diff
@@ -0,0 +1,160 @@
+  NSURL *bundleUrl = [_options respondsToSelector:@selector(jsCodeLocation)]
+    ? [_options jsCodeLocation]
+    : [[NSBundle bundleForClass:[self class]] URLForResource:@"hotels" withExtension:@"jsbundle"];
+  
+  self.view = [[RCTRootView alloc] initWithBundleURL:bundleUrl
+                                          moduleName:[_options moduleName]
+                                   initialProperties:[_options initialProperties]
+                                       launchOptions:nil];
+}
+
+- (void)viewDidDisappear:(BOOL)animated {
+  [super viewDidDisappear:animated];
+  
+  if ([self isBeingDismissed] || [self isMovingFromParentViewController]) {
+    [self notifyFinish];
+    [self setupReactWrappersWithObject:nil];
+    [self stopObservingGestures];
+  }
+}
+
+#pragma mark - Overriden methods
+
+- (BOOL)prefersStatusBarHidden {
+  return NO;
+}
+
+- (UIStatusBarStyle)preferredStatusBarStyle {
+  return UIStatusBarStyleLightContent;
+}
+
+- (UIStatusBarAnimation)preferredStatusBarUpdateAnimation {
+  return UIStatusBarAnimationFade;
+}
+
+#pragma mark - RNNavigation
+
+- (void)userWantsLeaveHotels {
+  [self notifyFinish];
+}
+
+#pragma mark - RNLogger
+
+- (void)ancillaryDisplayed:(NSString *)type {
+  if ([self didDisplayAncillary]) {
+    self.didDisplayAncillary(type);
+  }
+}
+
+- (void)ancillaryPurchased:(NSString *)type {
+  if ([self didPurchaseAncillary]) {
+    self.didPurchaseAncillary(type);
+  }
+}
+
+#pragma mark - RNTranslationManager
+
+- (NSString *)translate:(NSString *)key {
+  return [self.translationProvider localizedStringWithKey:key] ?: key;
+}
+
+#pragma mark - RNCurrencyManager
+
+- (NSString *)formatAmount:(NSNumber *)amount toCurrency:(NSString *)currency {
+  return [self.currencyFormatter formattedPrice:amount withCurrency:currency];
+}
+
+#pragma mark - Helpers
+
+- (void)notifyFinish {
+  if ([self.flowDelegate respondsToSelector:@selector(RNKiwiViewControllerDidFinish:)]) {
+    [self.flowDelegate RNKiwiViewControllerDidFinish:self];
+  }
+}
+
+#pragma mark - Gesture controllers observers
+
+- (void)removeGestures {
+  if ([self.flowDelegate respondsToSelector:@selector(RNKiwiViewControllerDidStartControllingGestures:)]) {
+    [self.flowDelegate RNKiwiViewControllerDidStartControllingGestures:self];
+  }
+}
+
+- (void)addGestures {
+  if ([self.flowDelegate respondsToSelector:@selector(RNKiwiViewControllerDidStopControllingGestures:)]) {
+    [self.flowDelegate RNKiwiViewControllerDidStopControllingGestures:self];
+  }
+}
+
+- (void)startObservingGestures {
+  [[NSNotificationCenter defaultCenter]
+   addObserver:self
+   selector:@selector(removeGestures)
+   name:@"RNKiwiRemoveGestures"
+   object:[_options moduleName]];
+  
+  [[NSNotificationCenter defaultCenter]
+   addObserver:self
+   selector:@selector(addGestures)
+   name:@"RNKiwiAddGestures"
+   object:[_options moduleName]];
+}
+
+- (void)stopObservingGestures {
+  [[NSNotificationCenter defaultCenter]
+   removeObserver:self
```

Comment:
You can use `[[NSNotificationCenter defaultCenter] removeObserver:self];` to unsubscrive from all notifications. No need to write it many times.